compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEConvertToDma.cpp

```diff
@@ -304,23 +305,49 @@ LogicalResult rewriteAsDma(PackOrUnpackOp op, IRRewriter &rewriter) {
 
 };  // namespace
 
-class AMDAIEPackToDmaPass
-    : public impl::AMDAIEPackToDmaBase<AMDAIEPackToDmaPass> {
+class AMDAIEConvertToDmaPass
+    : public impl::AMDAIEConvertToDmaBase<AMDAIEConvertToDmaPass> {
  public:
   void getDependentDialects(DialectRegistry &registry) const override {
     registry.insert<tensor::TensorDialect, linalg::LinalgDialect,
                     IREE::LinalgExt::IREELinalgExtDialect, AMDAIEDialect>();
   }
 
-  AMDAIEPackToDmaPass() = default;
-  AMDAIEPackToDmaPass(const AMDAIEPackToDmaPass &pass){};
+  AMDAIEConvertToDmaPass() = default;
+  AMDAIEConvertToDmaPass(const AMDAIEConvertToDmaPass &pass){};
   void runOnOperation() override;
 };
 
-void AMDAIEPackToDmaPass::runOnOperation() {
+void AMDAIEConvertToDmaPass::runOnOperation() {
   MLIRContext *context = &getContext();
   IRRewriter rewriter(context);
 
+  // Convert all linalg.copy to iree_linalg_ext.pack ops. We then bootstrap the
+  // work done for lowering the pack op to dmas as the next step.
+  auto walkCopyToPackResult =
```

Comment:
I think it's better/cleaner to create a separate function for linalg.copy conversion.